src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -663,6 +663,109 @@ public int GetLength(MemoryPoolIterator end)
             }
         }
 
+        public static unsafe void FastByteCopy(byte[] srcArray, uint srcOffset, byte[] destArray, uint destOffset, uint byteCount)
+        {
+            Debug.Assert(srcOffset + byteCount <= srcArray.Length);
+            Debug.Assert(destOffset + byteCount <= destArray.Length);
+            
+            if (byteCount < 64)
+            {
+                // Buffer.BlockCopy is expensive for small copies.  Do it manually ourselves.
```

Comment:
I agree, I'm just saying that figuring out a better way is probably a longer discussion.  Do you want to have that discussion before we decide whether to take this?